client/src/components/TypeList.vue

```diff
@@ -28,17 +65,145 @@ export default Vue.extend({
+            hide-details
+            class="my-1 ml-3 typeCheckBox"
+          />
+          <v-spacer />
+          <v-tooltip
+            open-delay="100"
+            bottom
+          >
+            <template #activator="{ on }">
+              <v-btn
+                class="mr-1 hover-show-child"
+                icon
+                small
+                v-on="on"
+                @click="clickEdit(type)"
+              >
+                <v-icon
+                  small
+                >
+                  mdi-pencil
+                </v-icon>
+              </v-btn>
+            </template>
+            <span>Edit</span>
+          </v-tooltip>
+        </v-row>
+      </v-container>
     </div>
+    <v-dialog
+      v-model="showPicker"
+      width="350"
+    >
+      <div
+        class="typeEdit"
+      >
+        <v-card>
+          <v-card-title>
+            Editing Type
+            <v-spacer />
+            <v-spacer />
+            <v-btn
+              class="mr-1"
+              icon
+              small
+              @click="showPicker = false"
+            >
+              <v-icon
+                small
+              >
+                mdi-close
+              </v-icon>
+            </v-btn>
+          </v-card-title>
+          <v-card-subtitle class="my-0 py-0">
+            <v-container class="py-0">
+              <v-row>
+                {{ selectedType }}
+                <v-spacer />
+                <v-btn
+                  icon
+                  small
+                  @click="showPicker = false"
+                >
+                  <v-icon
+                    small
+                  >
+                    mdi-exit
+                  </v-icon>
+                </v-btn>
+              </v-row>
+            </v-container>
+          </v-card-subtitle>
+          <v-container class="pt-0">
+            <v-row dense>
+              <v-col>
+                <v-text-field
+                  v-model="editingType"
+                  label="Name"
+                />
+              </v-col>
+            </v-row>
+            <v-row
+              dense
+              align="center"
+            >
+              <v-col class="mx-2">
+                <v-color-picker
+                  v-model="editingColor"
+                  hide-inputs
+                />
+              </v-col>
+            </v-row>
+          </v-container>
+          <v-card-actions>
+            <v-spacer />
+            <v-btn
+              color="primary"
+              text
+              @click="acceptChanges"
+            >
+              OK
+            </v-btn>
+          </v-card-actions>
+        </v-card>
+      </div>
+    </v-dialog>
   </div>
 </template>
+
+<style lang='scss'>
+.typeEdit{
+  overflow: hidden;
+}
+.typeCheckBox{
+max-width: 80%;
+overflow:hidden;
+}
+.hover-show-parent {
+  .hover-show-child {
+    display: none;
+  }
+  &:hover {
+    .hover-show-child {
+      display: inherit;
+    }
+  }
+}
```

Comment:
Lint this `SCSS`.  We need to add automated linting, but for now, just fixing the tabbing will do.